### PR TITLE
Remove flat routing

### PR DESCRIPTION
**Pull request by jbsv**
The original flat routing isn't used nor useable, so better take it off the code. To replace it, an option function has been added to tree routing constructor to be able to specify its height. Specifying a max tree height of 1 practically results in a flat routing.

### CODE REVIEW

File: mino/router/tree/mod.go [hunk 1]
```diff
@@ -32,8 +31,17 @@ type Router struct {
 	hsFac     router.HandshakeFactory
 }
 
+// RouterOption is the signature of the option constructor
+type RouterOption func(*Router)
+
+func WithHeight(maxHeight int) RouterOption {
```
Comment (jbsv) [inline mino/router/tree/mod.go:37]: @nkcr Is it possible to add a variable to an option function ?

Comment (nkcr) [reply mino/router/tree/mod.go:37]: yes, that's the point :)

Comment (jbsv) [reply mino/router/tree/mod.go:37]: Now that's magic to me!

Comment (pierluca) [reply mino/router/tree/mod.go:37]: ditto.

**Pull request merged**